src/Microsoft.ML.Data/EntryPoints/TransformModelImpl.cs

```diff
@@ -258,7 +262,8 @@ public Row GetRow(Row input, Func<int, bool> active)
                     _ectx.AssertValue(mapper);
                     mappers.Add(mapper);
                     actives.Add(activeCur);
-                    activeCur = mapper.GetDependencies(activeCur);
+                    var activeCurCol = mapper.GetDependencies(mapper.OutputSchema.Where(col => activeCur(col.Index)));
+                    activeCur = c => activeCurCol.Any(col => col.Index == c);
```

Comment:
>activeCurCol.Any(col => col.Index == c); [](start = 37, length = 40)

I don't like this usage of `Any` that I've been seeing... using quadratic algorithms is probably best avoided. Did we not have a utility method to take care of this predicate mapping problem? I think we did.

Reply:
`internal static Func<int, bool> FromColumnsToPredicate(IEnumerable<Schema.Column> columnsNeeded, Schema sourceSchema)`, you called it, in `RowCursorUtils`.

---
In reply to: [256593471](https://github.com/dotnet/machinelearning/pull/2504#discussion_r256593471) [](ancestors = 256593471)